### PR TITLE
doc: there is no case that is shown, so something was likely missing …

### PR DESCRIPTION
**Pull request by tshepang**
…from the change

### CODE REVIEW

**Comment by steveklabnik** [comment]
I am confused. It's showing what the value of this constant should be, given that we run the builders on this particular platform.

**Comment by tshepang** [comment]
Where is it shown that we are running on `unix/linux`?

**Comment by steveklabnik** [comment]
That's the idea with these. They're demonstrating the value this constant will have.

**Comment by tshepang** [comment]
But the constant depends on what platform was used to build the code, right?

**Comment by steveklabnik** [comment]
That's why it's "in this case". When these docs were built.

**Comment by tshepang** [comment]
I don't get it. There needs to be more explanation than just "in this case". More importantly, how is that info even important... why would we telling readers that the docs were built on `unix/linux`. I am also confused.

**Comment by steveklabnik** [comment]
Yeah, I didn't close it, because if its confusing, we should improve it.

These are string constants. This is intended to be showing the example value of said constant.

**Comment by MatejLach** [comment]
@steveklabnik If I understand @tshepang correctly, this link shows it in context:
https://doc.rust-lang.org/std/env/consts/ 

I think what @tshepang is saying is that when you look at the docs, it doesn't make sense to say `The family of the operating system. In this case, unix.` - instead just say 'The family of the operating system.' or something like `The family of the operating system. For example unix.`, because for a windows developer, it would be `The family of the operating system. In this case, windows.`

**Comment by tshepang** [comment]
@MatejLach understands me

**Comment by steveklabnik** [comment]
> because for a windows developer, it would be The family of the operating system. In this case, windows.

Yes. This existing text was a compromise; there's no way to directly insert it, so I put in the example from the way that the docs will be displayed officially.

**Comment by MatejLach** [comment]
@steveklabnik Just to make sure, you're saying that you inserted the `In this case, unix.` because the docs on the website will be running on a Linux box?

**Comment by steveklabnik** [comment]
Because the builder that builds the docs is running on a Linux box.

**Comment by frewsxcv** [comment]
For what it's worth, I think the text

>  In this case, `unix`.

is confusing since it's not immediately obvious what "this" is referring to. In my opinion, I don't think the existence of that sentence adds much value.

File: src/libstd/env.rs [hunk 1]
```diff
@@ -618,7 +618,7 @@ pub mod consts {
     #[stable(feature = "env", since = "1.0.0")]
     pub const ARCH: &'static str = super::arch::ARCH;
 
-    /// The family of the operating system. In this case, `unix`.
+    /// The family of the operating system.
```
Comment (mitaa) [inline src/libstd/env.rs]: `of the operating system in use`

That would align this with the doc comment below, and then it's clear to which `operating system` this actually refers to.

**Comment by mitaa** [comment]
These items / doc-comments are never cfg'd out, so it seems like if one where to build windows documentation the docs would still show `in this case ``unix`` ` instead of `in this case ``windows`` `

**Comment by steveklabnik** [comment]
@mitaa yes, that's what I meant by 

> This existing text was a compromise;

File: src/libstd/env.rs (hunk 1 above)
Comment (tshepang) [inline src/libstd/env.rs]: Adding "in use" is an improvement, thanks.

**Comment by tshepang** [comment]
@steveklabnik have you seen this page https://doc.rust-lang.org/nightly/std/env/consts?

**Comment by steveklabnik** [comment]
Yes, I have.

**Comment by steveklabnik** [comment]
So, to be clear, what I want to see in an improvement here is to make sure that we have examples of what this constants look like. This was how I did it; other ways are welcome. I don't want to remove the example, though.

**Comment by tshepang** [comment]
There is no example in that page. What example are you referring to?

**Comment by steveklabnik** [comment]
These are constants. This is an example showing the value of said constants.

**Comment by tshepang** [comment]
You are talking about an example as if it exists...

> I don't want to remove the example, though.

**Comment by steveklabnik** [comment]
The example is the text you are trying to remove in this PR.

**Comment by MatejLach** [comment]
@tshepang I think you're misunderstanding what @steveklabnik means by an example in this case. 
I think what Steve means by an example in this case *is* the `unix`.
@steveklabnik That's why I proposed changing it from `In this case, unix` to `for example, unix`

**Comment by ollie27** [comment]
There's a list of examples under the "Some possible values:" header. Is that not enough?

**Comment by tshepang** [comment]
They should be enough, but they don't render here. Maybe a bug in rustdoc?

**Comment by mitaa** [comment]
No, thats actually intentional. On module pages only short summaries are shown.
The full text is on the page of the const itself.
https://doc.rust-lang.org/nightly/std/env/consts/constant.ARCH.html

**Comment by tshepang** [comment]
Wow, wasn't even aware those are click-able. Well, that makes the case for this change even stronger, don't you think @steveklabnik?

**Comment by tshepang** [comment]
@steveklabnik ^

**Comment by ranma42** [comment]
I agree with @steveklabnik that providing an example value in the short summary is convenient, but I also think that @MatejLach is right: the wording `for example` would be more clear, especially considered that the examples are not generated based on what environment is building the docs.

Also, `ARCH` has no example in the short summary, which feels like an inconsistency when reading the https://doc.rust-lang.org/std/env/consts/ page. Would it be ok to add an example for it?

**Comment by steveklabnik** [comment]
@tshepang sorry, I did not see this since @aturon was the reviewer, not me. Let's fix that:

r? @steveklabnik

>  the wording for example would be more clear, 

Yes, this is what I'd like to see.

> Would it be ok to add an example for it?

Yes, though I'm guessing the examples would move into the body, rather than be a summary.

**Comment by tshepang** [comment]
PR updated

**Pull request merged**